### PR TITLE
added search, added a component that will be displayed if there is no…

### PR DESCRIPTION
**Pull request by Rashevskyyy**
… result, the message from the component has been translated into all languages ​​supported by the application

### CODE REVIEW

**Review by SkrypnikMR — APPROVED** [review]

**Comment by SkrypnikMR** [comment]
Perfect!

**Review by Mozzi22 — APPROVED** [review]

File: client/src/Store/chat/reducer.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ export const initialState = {
   currentRoom: { room_id: null, room_name: '' },
   error: false,
   isLoading: false,
+  filterValue: '',
```
Comment (MaxShvachko) [inline client/src/Store/chat/reducer.js]: rename this field

filrerByRoomName: ""

File: client/src/Store/chat/selectors.js [hunk 2]
```diff
@@ -21,3 +21,7 @@ export const currentRoomName = createSelector(
   currentRoom,
   ({ room_name }) => room_name,
 );
+export const filterValue = createSelector(
+    chatStore,
+    ({ filterValue }) => filterValue,
```
Comment (MaxShvachko) [inline client/src/Store/chat/selectors.js]: rename field like in comment upper

File: client/src/components/Chat/ChatList/ChatList.jsx [hunk 3]
```diff
@@ -3,24 +3,26 @@ import PropTypes from 'prop-types';
 import Search from './Search';
 import ChatListItems from './ChatListItems';
 import { StChatList } from './styled';
+import SearchNoRes from './Search/SearchNoRes';
+import { support } from '/src/helpers/support';
 
-const ChatList = ({ rooms }) => {
-  return (
+const ChatList = ({ rooms, filterValue }) => {
+    rooms = filterValue ? support.filteredRooms(filterValue, rooms) : rooms;
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatList/ChatList.jsx]: this logic must be in selector and I think helper for this logic is unuseful

File: client/src/components/Chat/ChatList/ChatList.jsx [hunk 4]
```diff
@@ -3,24 +3,26 @@ import PropTypes from 'prop-types';
...
+    return (
     <StChatList>
       <Search />
-      {rooms?.map(room => (
+      {rooms ? rooms.map(room => (
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatList/ChatList.jsx]: check rooms length for logic in this line.

`rooms?.length > 0 ? ..... : <SearchNoRes/>`

File: client/src/components/Chat/ChatList/Search/Search.jsx [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
 import { INPUT_PLACEHOLDER_SEARCH } from '/src/constants/componentsСonsts.js';
-import Input from '../../../UI/Input';
+import Input from '/src/components/UI/Input';
 import { StSearch } from './styled';
 
-const Search = ({ changeSearchValue, searchValue }) => {
+const Search = ({ setValue, filterValue }) => {
   return ( 
     <StSearch>  
       { INPUT_PLACEHOLDER_SEARCH.map(el => (
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatList/Search/Search.jsx]: you don't need map in this case.

File: client/src/components/Chat/ChatList/Search/Search.jsx [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
       <Input
         id={el.id}
```
Comment (MaxShvachko) [inline client/src/components/Chat/ChatList/Search/Search.jsx]: pass your data like 

id="nameOfYourId" like example

File: client/src/helpers/support.js [hunk 7]
```diff
@@ -19,4 +19,8 @@ export const support = {
         });
         return messagesFolders;
     },
+    filteredRooms: (filterValue, rooms) => {
+        const result = rooms.filter(room => room.room_name.includes(filterValue));
+        return result.length === 0 ? null : result;
```
Comment (MaxShvachko) [inline client/src/helpers/support.js]: move this logic to selectors

in you selector must be logic
change for compare filterValue.toLowerCase() and room.room_name.toLowerCase()
filterValue ? rooms?.filter .... : rooms

**Pull request merged**